test/regressionTest.m

```diff
@@ -0,0 +1,57 @@
+function regressionTest(commitBefore,commitAfter)
+    % Produce graphical output for two commits
+
+    % Work only with a clean working tree
+    [status, cmdout] = system('git status --porcelain');
+    if ~isempty(cmdout)
+        error('regressionTest:treeNotClean','Working tree is not clean.')
+    end
```

Comment:
While technically correct, let's not take such drastic measures. In git itself, all operations will warn when the tree is dirty. We should do the same. At most, I think we can show a message akin to "please clean your working copy using `git reset --hard`".